subprojects/core/src/main/java/org/gradle/internal/service/scopes/GlobalScopeServices.java

```diff
@@ -115,13 +116,15 @@
 public class GlobalScopeServices extends WorkerSharedGlobalScopeServices {
 
     private final GradleBuildEnvironment environment;
+    private final AgentStatus agentStatus;
 
-    public GlobalScopeServices(final boolean longLiving) {
-        this(longLiving, ClassPath.EMPTY);
+    public GlobalScopeServices(final boolean longLiving, AgentStatus agentStatus) {
```

Comment:
🎨 I'd also introduce a `GlobalScopeServices(boolean longLiving) { this(longLiving, AgentStatus.disabled()); }` to avoid spreading the knowledge of `AgentStatus.disabled()` to callers that don't care.